src/queryinterface.js

```diff
@@ -118,59 +139,96 @@ QueryInterface.prototype.$clearQueue = function (options) {
 };
 QueryInterface.prototype.$queueClear = QueryInterface.prototype.$clearQueue;
 
-/**
- * This is the mock method for getting results from the `QueryInterface`. This function
- * will get the next result in the queue and return that wrapped in a promise.
- * 
- * @instance
- * @param {Object} [options] Options used for this query
- * @param {Function} [options.fallbackFn] A fallback function to run if there are no results queued
- * @param {Boolean} [options.includeCreated] Flag indicating if a `created` value should be returned with the result for this query. Defaults to false
- * @param {Boolean} [options.includeAffectedRows] Flag indicating if the query expects `affectedRows` in the returned result parameters. Defautls to false
- * @param {Boolean} [options.stopPropagation] Flag indicating if result queue propagation should be stopped on this query. Defaults to false
- * @return {Promise} resolved or rejected promise from the next item in the review queue
- **/
-QueryInterface.prototype.$query = function (options) {
-	options = options || {};
-	
-	var fallbackFn = options.fallbackFn || this.options.fallbackFn;
-	
-	if(this._results.length) {
-		var result = this._results.shift();
-		
+function resultsQueueHandler(qi, options) {
```

Comment:
The contract of a handler is:

- Call `done()` if the handler can't process this request
- Call `done(<resolved promise>)` if the handler can process it with a valid value.
- Call `done(<rejected promise>)` if the handler can process it and the user wants to simulate a failure from Sequelize.
- Throw an error if there is an error in the handler.

Calling `done(<any value>)` is equivalent to `done(bluebird.resolve(val))`. This conversion happens at the end of `$query()`